### PR TITLE
[5.1] Convert mod_loginsupport to service provider

### PR DESCRIPTION
**Pull request by joomlaweby**
### Summary of Changes

Converts the login support module to service provider.

### Testing Instructions

1. Make sure you have administrator module mod_loginsupport published and configured
2. You should see module output on administrator login page

mod_loginsupport

### Actual result BEFORE applying this Pull Request

Module works as expected

### Expected result AFTER applying this Pull Request

Module works same way as before applying pull request

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/modules/mod_loginsupport/services/provider.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  mod_loginsupport
+ *
+ * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
```
Comment (richard67) [inline administrator/modules/mod_loginsupport/services/provider.php]:
```suggestion
\defined('_JEXEC') or die;
```
Fix PHPCS failing in Drone.

Comment (joomlaweby) [reply administrator/modules/mod_loginsupport/services/provider.php]: @richard67  Of course I can fix it, but I am confused, how to do it correctly. In different PR I did it this way - with \defined , but I was asked to remove \ from native functions in non-namespaced php files - https://github.com/joomla/joomla-cms/pull/42791#discussion_r1487197285, https://github.com/joomla/joomla-cms/pull/42791#discussion_r1487463780 .

In some other modules defines are with \

https://github.com/joomla/joomla-cms/blob/6c636c9db4528c8305d1911cbb8237e468a848f4/administrator/modules/mod_guidedtours/services/provider.php#L11

https://github.com/joomla/joomla-cms/blob/6c636c9db4528c8305d1911cbb8237e468a848f4/modules/mod_articles_latest/services/provider.php#L11

@richard67, @Quy and @sandewt , should I add \ in non-namespaced defines, so build will not fail, or remove, but build will fail?

Comment (sandewt) [reply administrator/modules/mod_loginsupport/services/provider.php]: Files **provider.php**:

Joomla 4.x
`defined('_JEXEC') or die;`

Joomla 5.x
`\defined('_JEXEC') or die;`

I was not aware of this difference.

To be consistent, I would here (in J5) use the code `\defined('_JEXEC') or die;`
However, I don't know the reason.

Comment (richard67) [reply administrator/modules/mod_loginsupport/services/provider.php]: @joomlaweby In the 5.0-dev branch (and so also 5.1-dev and so on) there was made a change, see PR #41427 . I remember it has been discussed in the maintainer team and it was decided to use the backslash in general, also for non namespaced files. So I would rely on the check in drone regarding this.

Comment (joomlaweby) [reply administrator/modules/mod_loginsupport/services/provider.php]: OK, great! Thanks for explanation! :-) Now it is fixed

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on 2c3a20accc33a2d2572f1018c178b661c5f4cdf7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42827">issues.joomla.org/tracker/joomla-cms/42827</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 2c3a20accc33a2d2572f1018c178b661c5f4cdf7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42827">issues.joomla.org/tracker/joomla-cms/42827</a>.</sub>

**Comment by eddiekonczal** [comment]
I have tested this item :white_check_mark: successfully on 2c3a20accc33a2d2572f1018c178b661c5f4cdf7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42827">issues.joomla.org/tracker/joomla-cms/42827</a>.</sub>

**Comment by LadySolveig** [comment]
Thank you @joomlaweby and also for testing @fgsw @viocassel @eddiekonczal

**Comment by joomlaweby** [comment]
Many thanks for reviewing and testing to @fgsw @viocassel @eddiekonczal @LadySolveig !

**Comment by richard67** [comment]
> Many thanks for reviewing and testing to @fgsw @viocassel @eddiekonczal @LadySolveig !

And to me not?

**Comment by LadySolveig** [comment]
Special thanks to @richard67 for your enduring support! 💯

**Pull request merged**